agent/agent.go

```diff
@@ -254,6 +254,9 @@ func (a *Agent) populateEC2InstanceID(ctx context.Context) {
 	a.ec2InstanceID = instanceID
 }
 
+// loop is responsible for continually polling for new tasks and processing them at.
+// If there are no more tasks, it sleeps at interval agentSleepInterval with jitteredSleep
```

Comment:
I don't think this comment needs to be so specific, because the internal sleep implementation between tasks when there is none available is not that critical to understanding its behavior of processing tasks. I would omit the second sentence.